modules/imaging_uploader/jsx/ImagingUploader.js

```diff
@@ -34,43 +36,171 @@ class ImagingUploader extends React.Component {
+   * @param {array} rowHeaders - array of table headers (column names)
+   *
+   * @return {*} a formatted table cell for a given column
+   */
+  formatColumn(column, cell, rowData, rowHeaders) {
+    // If a column if set as hidden, don't display it
+    if (this.state.hiddenHeaders.indexOf(column) > -1) {
+      return null;
+    }
+
+    // Create the mapping between rowHeaders and rowData in a row object.
+    let row = {};
+    rowHeaders.forEach((header, index) => {
+      row[header] = rowData[index];
+    });
+
+    // Default cell style
+    const cellStyle = {whiteSpace: 'nowrap'};
+
+    if (column === 'Progress') {
+      if (cell === 'Failure') {
+        cellStyle.color = '#fff';
+        return (
+          <td className="label-danger" style={cellStyle}>
+            {cell}
+          </td>
+        );
+      }
+
+      if (cell === 'In Progress...') {
+        cellStyle.color = '#fff';
+        return (
+          <td className="label-warning" style={cellStyle}>
+            {cell}
+          </td>
+        );
+      }
+
+      const created = row['Number Of MincCreated'];
+      const inserted = row['Number Of MincInserted'];
+      return (
+        <td style={cellStyle}>
+          {cell} ({inserted} out of {created})
+        </td>
+      );
+    }
+
+    if (column === 'Tarchive Info') {
+      if (!cell || cell === "0") {
+        return (<td></td>);
+      }
+
+      const url = loris.BaseURL + '/dicom_archive/viewDetails/?tarchiveID=' + cell;
       return (
-        <button className="btn-info has-spinner">
-          Loading
-          <span
-            className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
-          </span>
-        </button>
+        <td style={cellStyle}>
+          <a href={url}>View Details</a>
+        </td>
       );
     }
 
+    if (column === 'Number Of MincInserted') {
+      if (cell > 0) {
+        return (
+          <td style={cellStyle}>
+            <a onClick={this.handleClick.bind(null, row.CandID)}>{cell}</a>
+          </td>
+        );
+      }
+    }
+
+    if (column === 'Number Of MincCreated') {
+      let violatedScans;
+      if (row['Number Of MincCreated'] - row['Number Of MincInserted'] > 0) {
+        let numViolatedScans =
+             row['Number Of MincCreated'] - row['Number Of MincInserted'];
+
+        let patientName = row.PatientName;
+        violatedScans = <a onClick={this.openViolatedScans.bind(null, patientName)}>
+           ({numViolatedScans} violated scans)
+         </a>;
+      }
+
+      return (
+        <td style={cellStyle}>
+          {cell}
+          &nbsp;
+          {violatedScans}
+        </td>
+      );
+    }
+
+    return (<td style={cellStyle}>{cell}</td>);
+  }
+
+  /**
+   * Handles clicks on 'Number Of MincInserted' cells
+   *
+   * @param {string} dccid - dccid
+   * @param {object} e - event info
+   */
+  handleClick(dccid, e) {
+    loris.loadFilteredMenuClickHandler('imaging_browser/', {
```

Comment:
this might not be a 'now' problem, but i am wondering how we would remove this use of the loris global variable?

Reply:
Not quite sure what this really does, but I can leave a todo note.